### ISSUE TITLE
It's not pointer network

### ISSUE THREAD

**Issue by zygmuntz** [devnag/tensorflow-pointer-networks#1]
The defining characteristic of a pointer network is that the output is the same size as input - specifically, outputs are indexes into the input sequence. 

In your example, the output is always two pointers, so the whole thing is not a pointer network as described in the paper, it's something else.

**Comment by devnag** [comment]
Your definition is correct -- but your understanding of the code is incomplete. I put little comment hints next to the code lines with the tensor dimensions to try to make this a bit more clear, following the index names in Vinyals 2015 (https://arxiv.org/pdf/1506.03134v1.pdf). See section 2.3 of that paper.

i/I represents the output index, while j/J represents the input index. You're talking about the j/J index, which is the input. However, you can have multiple outputs, multiple pointers, _each_ of which indexes over the entire input sequence. This is how u_{j}^{i} is defined in 2.3, which explicitly allows for multiple pointers. The other part is that the output indices (whether there is 1, or 2, or 15 pointers) are conditioned on all previous indices: this is the p(Ci | C1, . . . , Ci−1,P) term, where P represents the full input sequence.

If you look at the python code, L181, the variable "idx_distribution" is equivalent to Vinyals' softmax(u^{i}) variable; it's shaped B x J (batch by input) because you can train multiple batches at once, but J is the pointer to the input range that you're talking about. The code just goes ahead and retrieves two pointers, along with a bunch of dereferencing the pointer since you need to send that input data back into the RNN (ptr_output at L189).

**Comment by zygmuntz** [comment]
I was missing the additional dimension of multiple pointers. Thanks for the explanation.
